cosima_cookbook/querying.py

```diff
@@ -305,13 +338,17 @@ def _ncfiles_for_variable(
         )
 
     # check whether the results are unique
-    unique_files = set(os.path.basename(f.NCFile.ncfile) for f in ncfiles)
-    if len(unique_files) > 1:
-        warnings.warn(
-            f"Your query gets a variable from differently-named files: {unique_files}. "
-            "This could lead to unexpected behaviour! Disambiguate by passing "
-            "ncfile= to getvar, specifying the desired file."
-        )
+    for attr, val in attrs_unique.items():
+        unique_attributes = set()
+        for f in ncfiles:
+            if attr in f.NCVar.attrs:
+                unique_attributes.add(str(f.NCVar.attrs[attr]))
```

Comment:
This won't detect `attr` being absent in some (but not all) files. To detect this, we could add an `else` clause:
```python
        for f in ncfiles:
             if attr in f.NCVar.attrs:
                 unique_attributes.add(str(f.NCVar.attrs[attr]))
             else:
                 unique_attributes.add("")  # zero-length names is forbidden in NetCDF, so this won't clash
```
`unique_attributes` should contain only `""` in cases that have no `cell_methods`, e.g. static files. If it contains more than that, we want to trigger the warning message.

Reply:
actually `unique_attributes.add(None)`  would probably make for a clearer warning message

Reply:
Done. Thanks.